### PR TITLE
PAXLOGGING-204: upgrade Log4j2

### PR DESCRIPTION
**Pull request by jvz**
There were a couple changes from 2.0 to 2.3 that had to be addressed here as well. If this needs to remain compatible with Java 1.6, then 2.3 is the latest version you can use. Otherwise, you can stay up to date (current: 2.4.1).

### CODE REVIEW

**Comment by ANierbeck** [comment]
looks ok to me, 
I just added you to the dev-team. Welcome aboard :-)

**Comment by jvz** [comment]
Thanks! I'll see what I can do to improve on the log4j2 support.

File: pax-logging-log4j2/src/main/java/org/apache/logging/log4j/core/config/plugins/util/ResolverUtil.java [hunk 1]
```diff
@@ -236,7 +235,7 @@ String extractPath(final URL url) throws UnsupportedEncodingException {
             // if URL-encoded file exists, don't decode it
             return urlPath;
         }
-        urlPath = URLDecoder.decode(urlPath, Charsets.UTF_8.name());
+        urlPath = URLDecoder.decode(urlPath, "UTF-8");
```
Comment (CodingFabian) [inline pax-logging-log4j2/src/main/java/org/apache/logging/log4j/core/config/plugins/util/ResolverUtil.java]: if its java 7+, we could do StandardCharsets.UTF_8.name()

Comment (jvz) [inline pax-logging-log4j2/src/main/java/org/apache/logging/log4j/core/config/plugins/util/ResolverUtil.java]: The pom file uses Java 6, not sure if that's still required.

On 3 December 2015 at 15:52, Fabian Lange <notifications@github.com> wrote:

> In
> pax-logging-log4j2/src/main/java/org/apache/logging/log4j/core/config/plugins/util/ResolverUtil.java
> <https://github.com/ops4j/org.ops4j.pax.logging/pull/19#discussion_r46618056>
> :
>
> > @@ -236,7 +235,7 @@ String extractPath(final URL url) throws UnsupportedEncodingException {
> >              // if URL-encoded file exists, don't decode it
> >              return urlPath;
> >          }
> > -        urlPath = URLDecoder.decode(urlPath, Charsets.UTF_8.name());
> > +        urlPath = URLDecoder.decode(urlPath, "UTF-8");
>
> if its java 7+, we could do StandardCharsets.UTF_8.name()
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ops4j/org.ops4j.pax.logging/pull/19/files#r46618056>.
>



-- 
Matt Sicker <boards@gmail.com>

**Pull request merged**